src/main.rs

```diff
@@ -224,32 +224,47 @@ fn handle_input_events(
         Err((Some(ENODEV), _)) => Ok(false),
         Err((_, error)) => Err(error).context("Error fetching input events"),
         Ok(events) => {
+            //A vector used to collect all mouse movement events
```

Comment:
nitpick: Since existing code uses the comment style `// comment`, could you consider putting a space (` `) after `//`? It's nice to have consistency across the codebase.